src/scenes/ObservationsRouter/details/index.tsx

```diff
@@ -174,9 +183,9 @@ export default function ObservationDetails(props: ObservationDetailsProps): JSX.
   }, [zoneNames, searchProps.filtersProps]);
 
   useEffect(() => {
-    if (matchResponse?.status === 'success') {
+    if (mergedMessage && matchResponse?.status === 'success') {
```

Comment:
If `mergedMessage` is only created when `onSaveMergedSpecies` runs and `merged.length > 0`, then maybe you can check that condition here? Also, are "merge" and "match" the same thing in this file?

Reply:
Merge and match are the same thing, yes. I'm checking `merged.length > 0` in `onSaveMergedSpecies` because if there are no merged species, there is no need to call the api.